### DISCUSSION TITLE
error importing fit files from sigma rox 4.0 firmware 2.x

### DISCUSSION THREAD

**Discussion by hubera1966** [mytourbook/mytourbook#958] [General]
Hello, i receive the followinf error importing the attachen fit file from sigma rox 4.0 ...

regards
[sigma_rox_4.zip](https://github.com/mytourbook/mytourbook/files/9896708/sigma_rox_4.zip)

<!DOCTYPE HTML>
<HTML style="margin: 0px; padding: 0px; width: 100%; height: 100%;"><HEAD>     
<STYLE>
html 
{
   overflow-y:          scroll;
}

body 
{
   font-size:           12px;
   color:               #ddd;
   background-color:    #333;
   xxxfont-weight:      100;  /* is not working */
}

html, body 
{
   margin:              5px;
}

p,td,ul 
{
   line-height:         130%;
}

p 
{
   margin:              0px;
}

img 
{
   border:              0;
}

table
{
   border-collapse:     collapse;
}

body,p,h1,h2,h3,h4,table,td,th,ul,ol,textarea,input 
{
   font-family:         sans-serif;       /* Segoe UI, helvetica, arial, */
}

a:LINK,
a:VISITED 
{
   color:               #fff;
   text-decoration:     none;
}

.icon
{
   width:                  16px; 
   height:                 16px; 
   min-width:              16px;          /* force width otherwise it can be smaller */

   background-repeat:      no-repeat;
   background-position-x:  5px;           /* this must be the same as .column otherwise it is not centered */
   xbackground-position-y: center;        /* prevent v-center when multiple lines are displayed */
}

.row
{
   vertical-align:      top;
}

.column
{
   padding-left:        5px;
}

.logItem
{
   white-space:         nowrap;
}

.subItem
{
   padding-left:        25px;
}

.title
{
   color:               #ABFE11;
   font-weight:         bold;
}

</STYLE>
 </HEAD><BODY><TABLE id="logs"><TBODY><TR class="row"><TD>20:29:03</TD><TD class="column logItem">[main - 1]</TD><TD class="column">1</TD><TD class="column icon"></TD><TD class="column  title">Touren Importieren</TD></TR><TR class="row"><TD>20:29:03</TD><TD class="column logItem">[main - 1]</TD><TD class="column">1</TD><TD class="column icon"></TD><TD class="column  title">Touren Importieren</TD></TR><TR class="row"><TD>20:29:05</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">2</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-info.png");'></TD><TD class="column  info">FIT SDK 21.84</TD></TR><TR class="row"><TD>20:29:05</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">3</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-info.png");'></TD><TD class="column subItem info">[FIT] C:\Users\huba\Dropbox\1_Dokumente_Briefe\gpx touren\2022-10-29_13-02-58_08B1C09C-087F-7F42-72C9-CFB18A312880_1035975778484_4.fit - File serial number is missing, device id cannot not be set</TD></TR><TR class="row"><TD>20:29:06</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">4</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-error.png");'></TD><TD class="column  logItem"><SPAN>java.lang.NullPointerException<BR>	at net.tourbook.data.TourData.lambda$3(TourData.java:7402)<BR>	at java.base/java.util.stream.ReferencePipeline$5$1.accept(Unknown Source)<BR>	at java.base/java.util.ArrayList$SubList$2.forEachRemaining(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source)<BR>	at java.base/java.util.stream.LongPipeline.toArray(Unknown Source)<BR>	at net.tourbook.data.TourData.finalizeTour_TimerPauses(TourData.java:7402)<BR>	at net.tourbook.device.garmin.fit.FitData.finalizeTour(FitData.java:216)<BR>	at net.tourbook.device.garmin.fit.FitDataReader.processDeviceData(FitDataReader.java:1062)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile_10(RawDataManager.java:2612)<BR>	at net.tourbook.importdata.RawDataManager.lambda$8(RawDataManager.java:2454)<BR>	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:57)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile(RawDataManager.java:2409)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromMultipleFiles_30_OneFile(RawDataManager.java:2273)<BR>	at net.tourbook.importdata.RawDataManager.lambda$6(RawDataManager.java:2242)<BR>	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)<BR>	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)<BR>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)<BR>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)<BR>	at java.base/java.lang.Thread.run(Unknown Source)<BR></SPAN></TD></TR><TR class="row"><TD>20:29:06</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">5</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-info.png");'></TD><TD class="column subItem info">[FIT] C:\Users\huba\Dropbox\1_Dokumente_Briefe\gpx touren\2022-10-29_13-02-58_08B1C09C-087F-7F42-72C9-CFB18A312880_1035975778484_4.fit - File serial number is missing, device id cannot not be set</TD></TR><TR class="row"><TD>20:29:06</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">6</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-error.png");'></TD><TD class="column  logItem"><SPAN>java.lang.NullPointerException<BR>	at net.tourbook.data.TourData.lambda$3(TourData.java:7402)<BR>	at java.base/java.util.stream.ReferencePipeline$5$1.accept(Unknown Source)<BR>	at java.base/java.util.ArrayList$SubList$2.forEachRemaining(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)<BR>	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source)<BR>	at java.base/java.util.stream.LongPipeline.toArray(Unknown Source)<BR>	at net.tourbook.data.TourData.finalizeTour_TimerPauses(TourData.java:7402)<BR>	at net.tourbook.device.garmin.fit.FitData.finalizeTour(FitData.java:216)<BR>	at net.tourbook.device.garmin.fit.FitDataReader.processDeviceData(FitDataReader.java:1062)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile_10(RawDataManager.java:2612)<BR>	at net.tourbook.importdata.RawDataManager.lambda$8(RawDataManager.java:2496)<BR>	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:57)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromOneFile(RawDataManager.java:2409)<BR>	at net.tourbook.importdata.RawDataManager.importTours_FromMultipleFiles_30_OneFile(RawDataManager.java:2273)<BR>	at net.tourbook.importdata.RawDataManager.lambda$6(RawDataManager.java:2242)<BR>	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)<BR>	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)<BR>	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)<BR>	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)<BR>	at java.base/java.lang.Thread.run(Unknown Source)<BR></SPAN></TD></TR><TR class="row"><TD>20:29:07</TD><TD class="column logItem">[Importing tours - 53]</TD><TD class="column">7</TD><TD class="column icon" style='background-image: url("file:/C:/Users/huba/.eclipse/1027998707_win32_win32_x86_64/configuration/org.eclipse.osgi/8/0/.cp/icons/state-error.png");'></TD><TD class="column subItem logItem">3,408 s ∙[Import fehlgeschlagen] C:\Users\huba\Dropbox\1_Dokumente_Briefe\gpx touren\2022-10-29_13-02-58_08B1C09C-087F-7F42-72C9-CFB18A312880_1035975778484_4.fit</TD></TR><TR class="row"><TD>20:29:07</TD><TD class="column logItem">[main - 1]</TD><TD class="column">8</TD><TD class="column icon"></TD><TD class="column  logItem">Importiert in 4,303 s</TD></TR></TBODY></TABLE></BODY></HTML>

**Comment by FJBDev** [comment]
@hubera1966 

Thanks for all those details.
This issue has already been reported [here ](https://sourceforge.net/p/mytourbook/discussion/622811/thread/d52269ce04/)and I fixed it [here](https://github.com/mytourbook/mytourbook/commit/743ec8aa99fcb64a930b6cd337df939284f2211c#diff-534a2a297b709e7c3ed21b2c2c297d85b65784ff323fea8a449d9c17b9ddf635) for the [next version of MT.](https://github.com/mytourbook/mytourbook/blob/49c94a3cd33d6f38a36f36bc2fdcc3e431fb8435/info/release-notes/22.next.0-readme.txt#L62)

Frederic

**Reply by peorgpewrfowefpwefpwep** [reply]
Thanks for a very fast response and a great bike tour tool for everybody.

regards

**Reply by FJBDev** [reply]
You're welcome, I'm glad to read you're enjoying MT

**Comment by hubera1966** [comment]
Hallo, Import funktioniert jetzt prima, vielen Dank !!

lg

**Comment by hubera1966** [comment]
Hello, now import works fine, thanks !!!
